compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -359,6 +359,21 @@ void addGPUWarpReductionPassPipeline(OpPassManager &pm) {
   nestedModulePM.addPass(createCSEPass());
 }
 
+void addGPUDataTilingPasses(OpPassManager &pm) {
+  tileAndDistributeToWorkgroup(pm);
+  auto &nestedModulePM = pm.nest<ModuleOp>();
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      createVectorizePackUnPackOpsPass());
+  addBufferizePasses(nestedModulePM);
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      createRemoveSingleIterationLoopPass());
+  nestedModulePM.addNestedPass<func::FuncOp>(createLLVMGPUTileAndDistribute());
```

Comment:
can we do distribution before bufferization. This pass is deprecated, you should be able to use the tiling on tensor path

Reply:
I'm actually confused that if the pass works. I should look into if the IR changes or not.  I'm throwing the PR out for some feedback. I'll do some more study and we can discuss it in our 1:1, thanks!

Reply:
makes sense, it might be worth adding a pipeline test to check the IR generated.